### PR TITLE
Added ansible to the rhel 8.4 distribution files.

### PR DESCRIPTION
**Pull request by AaronH88**
Added Ansible to the list of available packages for anyone building rhel 8.4 images.

Jira ticket: https://issues.redhat.com/browse/COMPOSER-981 

In order to keep CentOS 8 Streams as close as possible to rhel, ansibile has also been added to CentOS. 

TODO: 

- [x]  Test

### CODE REVIEW

File: distributions/rhel-84.json [hunk 1]
```diff
@@ -12,6 +12,9 @@
     }, {
       "baseurl": "https://cdn.redhat.com/content/dist/rhel8/8/x86_64/appstream/os",
       "rhsm": true
+    }, {                                                                                                                                                      
+      "baseurl": "https://cdn.redhat.com/content/dist/layered/rhel8/x86_64/ansible/2.9/os",                                                                   
+      "rhsm": true
```
Comment (Gundersanne) [inline distributions/rhel-84.json:17]: How about we add a field here `"requires_subscription": [ 'SKUxxxxx', 'ansible', etc... ]`. If present image-builder will check if the logged in user has access to these, and adds the packages (which should be located in a separate file), when querying the `/packages` endpoint. That way the user can search for/select them in the UI.

 The separate packages file should have a predictable name I think. Something like

`rhel-84-x86_64-packages.$SKU.json` with $SKU corresponding to the subscription name.

I'm not sure if we'll use SKU numbers or a name (for example `ansible`), whatever is easiest, both should be treated as strings anyway.

**Review by Gundersanne — DISMISSED** [review]
So by adding it to the main rhel84 packages file the result is that the ansible packages will seem to be available to anyone wanting to build the rhel-84 image. I don't think that's the intention?

If I understood it correctly, the aim is to add this repository and packages depending on whether or not the user has a valid ansible subscription.

So this will require code changes as well, to `internal/server/distribution.go` specifically.

File: distributions/rhel-84.json (hunk 1 above)
Comment (Gundersanne) [inline distributions/rhel-84.json:17]: Hmm when I enable it on a rhel-8.4 vm (`subscription-manager repos --enable ansible-2-for-rhel-8-x86_64-rpms`) I get:

```Repo ID:   ansible-2-for-rhel-8-x86_64-rpms
Repo Name: Red Hat Ansible Engine 2 for RHEL 8 x86_64 (RPMs)
Repo URL:  https://cdn.redhat.com/content/dist/layered/rhel8/x86_64/ansible/2/os
Enabled:   1
```

Is there a reason to pin it to `2.9?`

**Review by Gundersanne — CHANGES_REQUESTED** [review]
Thank you! Dismissed my earlier review as the requirements changed.

I have one question, and this will need a rebase to make the tests work  (due to https://github.com/osbuild/image-builder/commit/924a2d01b74609435569b45af2c8fc0698368cd4).

**Comment by teg** [comment]
> So by adding it to the main rhel84 packages file the result is that the ansible packages will seem to be available to anyone wanting to build the rhel-84 image. I don't think that's the intention?

Turns out this was a misunderstanding on my part. ansible-2.9 is available to anyone with a RHEL subscription.

@AaronH88 should we enable the same for CentOS Stream 8 though, do the equivalent ansible repos exist there?

**Comment by teg** [comment]
(C8S could be a separate PR, of course)

**Review by teg — APPROVED** [review]
Looking good!

File: distributions/rhel-84.json (hunk 1 above)
Comment (AaronH88) [reply distributions/rhel-84.json]: Agreed and changed

**Review by teg — DISMISSED** [review]

**Review by Gundersanne — APPROVED** [review]

**Pull request merged**